packages/core/components/DateRangePicker/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { DatePicker, Icon, IconButton } from "@fluentui/react";
+import * as React from "react";
+import FileFilter from "../../entity/FileFilter";
+import { DATE_ABSOLUTE_MIN, END_OF_TODAY } from "../../constants";
+import { extractDatesFromRangeOperatorFilterString } from "../../entity/AnnotationFormatter/date-time-formatter";
+
+import styles from "./DateRangePicker.module.css";
+
+interface DateRangePickerProps {
+    onSearch: (filterValue: string) => void;
+    onReset: () => void;
+    currentRange: FileFilter | undefined;
+}
+
+// Color chosen from App.module.css
+const PURPLE_ICON_STYLE = { icon: { color: "#827aa3" } };
+
+// Because the datestring comes in as an ISO formatted date like 2021-01-02
+// creating a new date from that would result in a date displayed as the
+// day before due to the UTC offset, to account for this we can add in the offset
+// ahead of time.
+export function extractDateFromDateString(dateString?: string): Date | undefined {
+    if (!dateString) {
+        return undefined;
+    }
+    const date = new Date(dateString);
+    date.setMinutes(date.getTimezoneOffset());
+    return date;
+}
+
+/**
+ * This component renders a simple form for selecting a minimum and maximum date range
+ */
+export default function DateRangePicker(props: DateRangePickerProps) {
+    const { onSearch, onReset, currentRange } = props;
+
+    function onDateRangeSelection(startDate: Date | null, endDate: Date | null) {
+        // Derive previous startDate/endDate from current filter state, if possible
+        let oldStartDate;
+        let oldEndDate;
+        const splitFileAttributeFilter = extractDatesFromRangeOperatorFilterString(
```

Comment:
Looks like each time this function is used it then splits the dates into a start and end. Seems like a simpler pattern would be to have the function return undefined values for the start and end date keys so that you can always use object destructuring to yank the start and end date values our and then do any processing you need to on those dates.

Reply:
updated in [e18657f](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/71/commits/e18657f1d8fc7e7a7a585dfbb8e3110f9dbe5eb6)